R/utils.R

```diff
@@ -141,7 +141,7 @@ JS <- function(...) {
 #   JavaScript are to be identified
 # @author Yihui Xie
 JSEvals <- function(list) {
-  evals <- names(which(unlist(shouldEval(list))))
+  evals <- names(which(unlist(shouldEval(list)))) %||% list()
```

Comment:
I think this should have a comment explaining why `list()` is there; otherwise it'll probably be forgotten.